### ISSUE TITLE
Fixture not captured when escaping regular expressions

### ISSUE THREAD

**Issue by psimm** [ropensci/dittodb#150]
### Brief description of the problem

I'm testing a query that uses `filter(str_detect(columnname, "\\."))`. The pattern `\\.` is a regular expression that looks for a literal `.` and uses `\\` to escape it both in R and in Regex. When this pattern is present in the query, the test fixture is not captured and the test fails. It also happens with other escaped patterns, e.g. `\\^`.

Could be related to #146 which enabled quoting. Maybe `\\` breaks the quoting?

### The kind of database backend you are trying to test

PostgresSQL 13.3 with package RPostgres

### Reprex

```r
library(DBI)
library(dittodb)
library(dplyr)
library(testthat)

db_connect <- function() {
  dbConnect(
    drv = RPostgres::Postgres(),
    host = "host",
    user = "user",
    password = "password",
    dbname = "dbname"
  )
}

dbWriteTable(
  conn = db_connect(),
  name = "test",
  value = data.frame(label = c(".a", "a", "b", "b"), value = c(1, 2, 3, 4)),
  overwrite = TRUE
)

myfun <- function(conn) {
  tbl(conn, "test") %>% 
    filter(str_detect(label, "\\.")) %>% 
    collect()
}

start_db_capturing()
conn <- db_connect()
act <- myfun(conn)
dbDisconnect(conn)
stop_db_capturing()

with_mock_db({
  test_that("myfun works", {
    conn <- db_connect()
    act <- myfun(conn)
    dbDisconnect(conn)
    expect_true(is.data.frame(act))
  })
})
```

Backtrace:

```

── Error (Line 4): myfun works ─────────────────────────────────────────────────
Error: Couldn't find the file psimm/SELECT-fa7d7d.R in any of the mock directories.
Backtrace:
  1. global::myfun(conn)
  4. dbplyr:::collect.tbl_sql(.)
  6. dbplyr:::db_collect.DBIConnection(x$src$con, sql, n = n, warn_incomplete = warn_incomplete)
 10. dittodb::dbFetch(res, n = n)
 13. dittodb:::find_file(path)
```

`sessionInfo()`
```
R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 11.5.2

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] testthat_3.0.4     dplyr_1.0.7        dittodb_0.1.3.9000 DBI_1.1.1         

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.7       pillar_1.6.2     compiler_4.1.0   dbplyr_2.1.1     RPostgres_1.3.3  tools_4.1.0     
 [7] digest_0.6.27    bit_4.0.4        pkgload_1.2.1    lubridate_1.7.10 RSQLite_2.2.8    memoise_2.0.0   
[13] lifecycle_1.0.0  tibble_3.1.4     pkgconfig_2.0.3  rlang_0.4.11     cli_3.0.1        rstudioapi_0.13 
[19] fastmap_1.1.0    withr_2.4.2      generics_0.1.0   desc_1.3.0       vctrs_0.3.8      hms_1.1.0       
[25] bit64_4.0.5      rprojroot_2.0.2  tidyselect_1.1.1 glue_1.4.2       R6_2.5.1         fansi_0.5.0     
[31] purrr_0.3.4      blob_1.2.2       magrittr_2.0.1   ellipsis_0.3.2   assertthat_0.2.1 renv_0.14.0     
[37] utf8_1.2.2       cachem_1.0.6     crayon_1.4.1
```

**Comment by jonkeane** [comment]
Ok, I think I've addressed this in the same branch/PR in #151. Would you mind trying that and seeing if it resolves both issues? 

Also, you're not using odbc as well are you? It turns out the fix for this / dbplyr translations for odbc aren't quite right, but I can do that as a followup to this one so that I can get you unblocked.

**Comment by psimm** [comment]
Both issues are solved, thanks a lot! Tested it with the reprexes and on my actual use case. 
No, I'm not using odbc.
